miden-tx/tests/test_miden_atomic_swap.rs

```diff
@@ -0,0 +1,86 @@
+use common::{
+    get_account_with_default_account_code, get_new_key_pair_with_advice_map, MockDataStore,
+};
+use miden_lib::notes::{create_note, Script};
+use miden_objects::{
+    accounts::{Account, AccountId, AccountVault, ACCOUNT_ID_FUNGIBLE_FAUCET_ON_CHAIN},
+    assembly::ProgramAst,
+    assets::{Asset, FungibleAsset},
+    Felt,
+};
+use miden_tx::TransactionExecutor;
+use mock::constants::{ACCOUNT_ID_REGULAR_ACCOUNT_UPDATABLE_CODE_ON_CHAIN, ACCOUNT_ID_SENDER};
+use vm_core::StarkField;
+
+mod common;
+
+#[test]
+fn test_atomic_swap_script() {
+    // Create assets
+    let faucet_id = AccountId::try_from(ACCOUNT_ID_FUNGIBLE_FAUCET_ON_CHAIN).unwrap();
+    let fungible_asset: Asset = FungibleAsset::new(faucet_id, 100).unwrap().into();
```

Comment:
You will need to create 2 assets for this test. The first asset will be on the note and the second asset will be with the target account. The target account then consumes the ASWAP script.